### PR TITLE
Heat Index Notebook

### PR DESCRIPTION
**Pull request by vicford**
This notebook walks through how to obtain and interpret heat index values. 

---
**To test**
Run through the notebook and let me know scientific + programmatic thougths!

### CODE REVIEW

**Review by claalmve — APPROVED** [review]

**Comment by claalmve** [comment]
I think the only change/reminder was to add the markdown warning about the runtime for downloading hourly data, but otherwise looks good to me!

**Comment by Tianchi-Liu** [comment]
Great to see heat index in action! I have one major concern about the non-local time that starts at about 4pm Pacific Time. While the counts of extreme "days" still capture the trends, their definition is less intuitive than if local hours are used. Plus the use of this non-local time would inflate the count of extreme days in step 2 (say a day with a few "very warm" hours that sit right across this non-local time zone's midnight, one extra "very warm" "day" might be counted compared to when the local time zone is used).

File: collaborative/IOU/heat_index.ipynb [hunk 1]
```diff
@@ -0,0 +1,557 @@
...
+    "print('The projected change in the median Heat Index value from historical is: {:.2f}°F'.\n",
+    "      format(hi_proj - hi_hist))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2f4e4b46-5312-47e5-99fc-95fec3d828cc",
+   "metadata": {},
+   "source": [
+    "### Step 2: Calculate the number of days each year above a Heat Index threshold\n",
```
Comment (Tianchi-Liu) [inline collaborative/IOU/heat_index.ipynb:345]: Nit: ## would suffice, same for Step 3

File: collaborative/IOU/heat_index.ipynb [hunk 2]
```diff
@@ -0,0 +1,557 @@
...
+   "id": "09fba876-5f97-4262-b80e-a5ceb226e8db",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "selections.latitude = (stn_lat - 0.05, stn_lat + 0.05)\n",
+    "selections.longitude = (stn_lon - 0.05, stn_lon + 0.05)\n",
+    "\n",
+    "# because we're retrieving a single grid cell, we also take the area average\n",
```
Comment (Tianchi-Liu) [inline collaborative/IOU/heat_index.ipynb:150]: could be multiple grid cells (e.g. Burbank-Glendale-Pasadena Airport (KBUR)), which is no problem. The values would just be weighted area averages.

**Review by Tianchi-Liu — COMMENTED** [review]
The later comments on the notebook are minor compared to the time zone one. They don't have to be addressed as I understand the notebook is still under development!

**Comment by vicford** [comment]
> I have one major concern about the non-local time that starts at about 4pm Pacific Time. While the counts of extreme "days" still capture the trends, their definition is less intuitive than if local hours are used. Plus the use of this non-local time would inflate the count of extreme days in step 2 (say a day with a few "very warm" hours that sit right across this non-local time zone's midnight, one extra "very warm" "day" might be counted compared to when the local time zone is used).

@Tianchi-Liu this is a really good point. Unfortunately, we don't have a clean way to do a local timezone conversion yet, and it's becoming increasingly more relevant for multiple notebooks -- I've got a trello card up for this issue. I have tested a "brute force" way but definitely need assistance on its improvement and implementation as a function we can call when needed. For now, I think as this notebook is still very preliminary and that the stakeholder in particular was most interested in the long-term trends over time between a projected future and the historical period, we would be fine for an approximation at present, and can work on refining the values for the next iterations of this notebook.

File: collaborative/IOU/heat_index.ipynb (hunk 2 above)
Comment (Tianchi-Liu) [reply collaborative/IOU/heat_index.ipynb]: get_closest_gridcell makes sense! I noticed the `selections.cached_area` is set to PG&E, so get_closest_gridcell wouldn't return the expected data if the input lat/lon is outside of the PG&E zone. If you think the stakeholders might do that for some reason, we'd need a workaround. 

This line below gives the PG&E stations you listed. Feel free to take or leave.
`wx_stns[wx_stns.station.str.contains('|'.join(['Arcata', 'Red Bluff', 'Stockton', 'San Jose', 'Fresno', 'San Luis Obisbpo', 'Bakersfield']))]`

**Review by Tianchi-Liu — APPROVED** [review]

**Pull request merged**